### PR TITLE
2022 Third parties Chapter

### LINKED ISSUE #2886: Third Parties 2022
# Third Parties 2022

Third Parties illustration

If you're interested in contributing to the Third Parties chapter of the 2022 Web Almanac, please reply to this issue and indicate which role or roles best fit your interest and availability: [author](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide), [reviewer](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Reviewers'-Guide), [analyst](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Analysts'-Guide), and/or [editor](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Editors'-Guide).

## Content team

| Lead | Authors | Reviewers | Analysts | Editors | Coordinator |
| ---- | ------- | --------- | -------- | ------- | ----------- |
| @imeugenia | @imeugenia | @tunetheweb @housseindjirdeh @pepelsbey @kevinfarrugia @alexnj  | @kevinfarrugia | - | @siakaramalegos  |

<details>
<summary><b>Expand for more information about each role 👀</b></summary>

- The **[content team lead](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Content-Team-Leads'-Guide)** is the chapter owner and responsible for setting the scope of the chapter and managing contributors' day-to-day progress.
- **[Authors](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide)** are subject matter experts and lead the content direction for each chapter. Chapters typically have one or two authors. Authors are responsible for planning the outline of the chapter, analyzing stats and trends, and writing the annual report.
- **[Reviewers](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Reviewers'-Guide)** are also subject matter experts and assist authors with technical reviews during the planning, analyzing, and writing phases.
- **[Analysts](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Analysts'-Guide)** are responsible for researching the stats and trends used throughout the Almanac. Analysts work closely with authors and reviewers during the planning phase to give direction on the types of stats that are possible from the dataset, and during the analyzing/writing phases to ensure that the stats are used correctly.
- **[Editors](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Editors'-Guide)** are technical writers who have a penchant for both technical and non-technical content correctness. Editors have a mastery of the English language and work closely with authors to help wordsmith content and ensure that everything fits together as a cohesive unit.
- The **[section coordinator](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Section-Leads'-Guide)** is the overall owner for all chapters within a section like "User Experience" or "Page Content" and helps to keep each chapter on schedule.

_Note: The time commitment for each role varies by the chapter's scope and complexity as well as the number of contributors._

For an overview of how the roles work together at each phase of the project, see the [Chapter Lifecycle](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle) doc.
</details>

### [**0. Form the content team**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#0-create-content-team)
+ [x] **May 1**: The content team has at least one author, reviewer, and analyst

### [**1. Plan content**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#1-plan-content)
+ [x] **May 15** The content team has completed the chapter outline in the [draft doc][~google-doc]

### [**2. Gather data**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#2-gather-data)
+ [x] **June 1**: Analysts have added all necessary [custom metrics](https://github.com/HTTPArchive/custom-metrics/blob/main/README.md) and drafted a PR ([example](https://github.com/HTTPArchive/almanac.httparchive.org/pull/1087)) to track query progress
- **June 1 - 15**: HTTP Archive runs the June crawl

### [**3. Validate results**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#3-validate-results)
+ [x] **August 1**: Analysts have queried all metrics and saved the output to the [results sheet][~google-sheets]

### [**4. Draft content**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#4-draft-content)
+ [x] **September 1**: The content team has written, reviewed, and edited the chapter in the [doc][~google-doc]

### [**5. Publication**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#5-publication)
+ [x] **September 15**: The completed chapter and all required [metadata](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide#metadata-to-add-at-the-top-of-your-chapters) and [figures](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Figures-Guide) are converted to markdown and submitted to [GitHub][~chapter-markdown]
- **September 26**: Target launch date 🚀

## Chapter resources

Refer to these 2022 Third Parties resources throughout the content creation process:

[📄 Google Docs][~google-doc] for outlining and drafting content
[🔍 SQL files][~sql-files] for committing the queries used during analysis
[📊 Google Sheets][~google-sheets] for saving the results of queries
[📝 Markdown file][~chapter-markdown] for publishing content and managing public metadata
[💬 #web-almanac-third-parties on Slack][~slack-channel] for team coordination

[~google-doc]: https://docs.google.com/document/d/1QB5AGi8ye21TqX8ySq7XIWODXZakJ903hIITne1IKOc/edit?usp=sharing
[~sql-files]: https://github.com/HTTPArchive/almanac.httparchive.org/tree/main/sql/2022/third-parties
[~google-sheets]: https://docs.google.com/spreadsheets/d/1YqoRRsyiNsrEabVLu2nRU98JIG_0zLLuoQhC2nX8xbM/edit?usp=sharing
[~chapter-markdown]: https://github.com/HTTPArchive/almanac.httparchive.org/tree/main/src/content/en/2022/third-parties.md
[~slack-channel]: https://join.slack.com/t/httparchive/shared_invite/zt-45sgwmnb-eDEatOhqssqNAKxxOSLAaA

### PR DESCRIPTION
**Pull request by imeugenia**
Closes #2886

Staged link: https://20220915t083540-dot-webalmanac.uk.r.appspot.com/en/2022/third-parties
Home page quotes: https://20220915t083540-dot-webalmanac.uk.r.appspot.com/en/2022/?feat=third-parties#featured-chapter

Some references:

[Draft Doc](https://docs.google.com/document/d/1QB5AGi8ye21TqX8ySq7XIWODXZakJ903hIITne1IKOc/edit#heading=h.qj2uj59tp25w)
[Chart Spreadsheet](https://docs.google.com/spreadsheets/d/1YqoRRsyiNsrEabVLu2nRU98JIG_0zLLuoQhC2nX8xbM/)

Remaining todos:

- [x] Update Prevalence section (data about 3P by rank)

### CODE REVIEW

File: src/content/en/2022/third-parties.md [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
-featured_quote: Third parties are often associated with negative impact on web performance. However, both third parties and website developers can eliminate it so that third-party functionality isn't harming user experience.
+featured_quote: Third parties are often associated with negative impact on web performance. However, both third-party and website developers can eliminate it so that third-party functionality isn't harming user experience.
 featured_stat_1: 94%
 featured_stat_label_1: Websites using at least one third party
-featured_stat_2: 45%
-featured_stat_label_2: Requests made by websites that are third-party requests.
-featured_stat_3: 34%
-featured_stat_label_3: Third-party requests serving script content.
+featured_stat_2: 60%
+featured_stat_label_2: Top 10 third parties are blocking main thread on more than 60% of websites
```
Comment (imeugenia) [inline src/content/en/2022/third-parties.md:15]: Took it from this one: https://20220915t083540-dot-webalmanac.uk.r.appspot.com/en/2022/third-parties#fig-7

Comment (tunetheweb) [reply src/content/en/2022/third-parties.md:16]: I disagree with this stat. Or at least the way it is worded currently.

60% of the websites—that load those third-parties—_may_ block the main thread on mobile if they are as low powered as the Lighthouse mobile client.

So a few things with this:
- It's not 60% of all websites as the above wording implies, but only 60% of websites that load each of those third parties. Many sites will not load any of those third-parties.
- We cannot definitely say that they are blocking the main thread as Lighthouse is a lab-based test and is run on throttled devices. It's entirely possible that the website that use those third parties are only visited by people with more powerful devices. That's unlikely of course but I worry about the definitiveness of saying they definitely are blocking the main thread.

Personally I think this stats is a little hard to explain as a featured stat given the nuances there. I think I'd pick another stat here. Or pick one of that this a little easier to explain (e.g. "90% - Percentage YouTube use which blocks main thread on mobile devices". Still a little definite for my taste but at least can point to an easy to understand audit that backs this up).

File: src/config/2022.json [hunk 2]
```diff
@@ -866,7 +865,8 @@
         "authors"
       ],
       "avatar_url": "17236402",
-      "github": "imeugenia"
+      "github": "imeugenia",
+      "twitter": "jevgeniazi"
```
Comment (tunetheweb) [inline src/config/2022.json:869]: @imeugenia I added your Twitter profile here but feel free to update to remove if you'd prefer not to have it. you can also add a website and LinkedIn profile if you so desire.

File: src/content/en/2022/third-parties.md [hunk 3]
```diff
@@ -1,22 +1,400 @@
...
+description: Third Parties chapter of the 2022 Web Almanac covering data of what third parties are usedon the web and a deep dive into preventing them causing negatibe performance impacts.
 authors: [imeugenia]
-reviewers: [tunetheweb, housseindjirdeh, pepelsbey, kevinfarrugia, alexnj]
-analysts: [kevinfarrugia, msolercanals]
-editors: []
+reviewers: [tunetheweb, kevinfarrugia, alexnj]
+analysts: [kevinfarrugia]
+editors: [shantsis]
 translators: []
+imeugenia_bio: Eugenia is a Senior frontend engineer at <a hreflang="en" href="https://gorillas.io/">Gorillas</a>. She is passionate about web performance and state machines.
```
Comment (tunetheweb) [inline src/content/en/2022/third-parties.md:10]: @imeugenia I filled in this bio for you, but feel free to update it to whatever you want.

**Review by tunetheweb — COMMENTED** [review]
I've given this chapter a good going through tonight and think, apart from one outstanding question on one of the stats, it's in a good place to merge and mark it as done! 🎉

@imeugenia could you have a look at that comment, and also a couple of other things I noted on your profile and bio?

Do let me know if you plan to change this chapter anymore but it's looking good to me!

**Comment by rviscomi** [comment]
Can we open this PR up for review?

File: src/content/en/2022/third-parties.md (hunk 1 above)
Comment (imeugenia) [reply src/content/en/2022/third-parties.md:16]: That is fair point, thank you! 
I want to choose something that shows the impact on performance but also would fit in one short sentence to explain.
How about this one:
```
featured_stat_2: 2 sec
featured_stat_label_2: Potential median render-blocking time caused by Google Maps
```

File: src/content/en/2022/third-parties.md (hunk 3 above)
Comment (imeugenia) [reply src/content/en/2022/third-parties.md]: Have updates my bio, thank!

File: src/content/en/2022/third-parties.md (hunk 1 above)
Comment (tunetheweb) [reply src/content/en/2022/third-parties.md]: LGTM!

**Review by tunetheweb — APPROVED** [review]

**Pull request merged**